src/middlewared/middlewared/plugins/failover_/disabled_reasons.py

```diff
@@ -22,24 +41,8 @@ class Config:
     @returns(List('reasons', items=[Str('reason')]))
     @pass_app()
     def reasons(self, app):
-        """
-        Returns a list of reasons why failover is not enabled/functional.
-
-        NO_VOLUME - There are no pools configured.
-        NO_VIP - There are no interfaces configured with Virtual IP.
-        NO_SYSTEM_READY - Other storage controller has not finished booting.
-        NO_PONG - Other storage controller is not communicable.
-        NO_FAILOVER - Failover is administratively disabled.
-        NO_LICENSE - Other storage controller has no license.
-        DISAGREE_VIP - Nodes Virtual IP states do not agree.
-        MISMATCH_DISKS - The storage controllers do not have the same quantity of disks.
-        MISMATCH_VERSIONS - TrueNAS software versions do not match between storage controllers.
-        NO_CRITICAL_INTERFACES - No network interfaces are marked critical for failover.
-        NO_FENCED - Zpools are imported but fenced isn't running.
-        LOC_FAILOVER_ONGOING - This node is currently processing a failover event.
-        REM_FAILOVER_ONGOING - Other node is currently processing a failover event.
-        NO_HEARTBEAT_IFACE - Local heartbeat interface does not exist.
-        NO_CARRIER_ON_HEARTBEAT - Local heartbeat interface is down.
+        """Returns a list of reasons why failover is not enabled/functional.
+        See `DisabledReasonsEnum` for the reasons and their explanation.
         """
         reasons = self.middleware.call_sync('failover.disabled.get_reasons', app)
         if reasons != FailoverDisabledReasonsService.LAST_DISABLED_REASONS:
```

Comment:
Probably fine (and orthogonal to the changes in the PR), but Ill just observe that this will fire on the change from `None` to the empty set (or non-empty set, as the case may be).